cmd/hipsterfy/Hipsterfy/Server.hs

```diff
@@ -74,20 +74,19 @@ handleCompare = post "/compare" $ do
   -- Check that the friend code is valid.
   friendCode <- param "friend-code"
   maybeFriend <- getUserByFriendCode friendCode
-  friend <- case maybeFriend of
-    Just f -> return f
-    -- TODO: display an error message for invalid friend codes.
-    Nothing -> redirect "/"
-
-  -- Load followed artists.
-  -- TODO: load these in parallel
-  yourArtists <- getFollowedArtists $ userID user
-  friendArtists <- getFollowedArtists $ userID friend
+  case maybeFriend of
+    Just f -> do 
+      -- Load followed artists.
+      -- TODO: load these in parallel
+      yourArtists <- getFollowedArtists $ userID user
+      friendArtists <- getFollowedArtists $ userID f
 
-  -- Render page.
-  yourStatus <- infoToStatus $ updateJobInfo user
-  friendStatus <- infoToStatus $ updateJobInfo friend
-  html $ comparePage (yourStatus, yourArtists) (friendStatus, friendArtists)
+      -- Render page.
+      yourStatus <- infoToStatus $ updateJobInfo user
+      friendStatus <- infoToStatus $ updateJobInfo f
+      html $ comparePage $ Right ((yourStatus, yourArtists), (friendStatus, friendArtists))
+    Nothing -> do html $ comparePage $ Left FriendCodeError 
```

Comment:
This should really just be its own page. It has literally nothing in common with the page rendered in the `Right` path.